framework/mesh/mesh_generator/extruder_mesh_generator.cc

```diff
@@ -101,17 +101,17 @@ ExtruderMeshGenerator::ExtruderMeshGenerator(const InputParameters& params)
   } // layer_block in layers_param
 }
 
-std::unique_ptr<UnpartitionedMesh>
-ExtruderMeshGenerator::GenerateUnpartitionedMesh(std::unique_ptr<UnpartitionedMesh> input_umesh)
+std::shared_ptr<UnpartitionedMesh>
```

Comment:
what are the consequences (or reasons) for going from a unique ptr to a shared ptr ?

Reply:
`UnpartitionedMesh`es are stored as `shared_ptr`. When mesh generators were executed there were some `std::move`s so that the `shared_ptr` could be passed into the API as a `unique_ptr`. If I understand correctly `std::move` invalidates the source memory, i.e. the stored `shared_ptr` would no longer valid. But, the lua layer might be keeping a lua handle into that stack, so when it would want to access it again (i.e. after the mesh generators were executed) the expected `UnpartitionedMesh` would not be there. This was my line of thinking (but I did not test it).

Reply:
ok. tagging @wdhawkins and @zhardy-lanl for awareness and maybe feedback

Reply:
I would agree with this assessment. Was there a specific reason (that you can tell) that we go from `shared_ptr` to `unique_ptr` for the mesh generator? I haven't looked very closely at this bit of the code.

Reply:
If there was a reason for `unique_ptr` I don't know it. If it isn't obvious why it is used, there is no reason to not move to the less restrictive `shared_ptr` IMO.

Reply:
This could be the reason for choosing `unique_ptr`:

https://github.com/idaholab/moose/blob/0814fb339a029de034bb8b2de55749ef6d8b3426/framework/include/meshgenerators/MeshGenerator.h#L69

and

https://github.com/idaholab/moose/blob/0814fb339a029de034bb8b2de55749ef6d8b3426/framework/include/meshgenerators/MeshGenerator.h#L265